manager/integration/tests/test_node.py

```diff
@@ -1308,3 +1310,150 @@ def test_node_config_annotations(client, core_api,  # NOQA
         break
+
+
+@pytest.mark.node
+def test_invalid_node_annotations(client, core_api,  # NOQA
+                                 reset_default_disk_label,  # NOQA
+                                 reset_disk_and_tag_annotations,  # NOQA
+                                 reset_disk_settings):  # NOQA
+
+    setting = client.by_id_setting(SETTING_CREATE_DEFAULT_DISK_LABELED_NODES)
+    client.update(setting, value="true")
+
+    nodes = client.list_node().data
+    node_name = nodes[0].id
+
+    # The invalid disk annotation shouldn't intervene the node controller.
+    cleanup_node_disks(client, node_name)
+
+    core_api.patch_node(node_name, {
+        "metadata": {
+            "labels": {
+                CREATE_DEFAULT_DISK_LABEL:
+                    CREATE_DEFAULT_DISK_LABEL_VALUE_CONFIG
+            },
+            "annotations": {
+                DEFAULT_DISK_CONFIG_ANNOTATION:
+                    '[{"path":"/invalid-path","allowScheduling":false,' +
+                    '"storageReserved":1024,"tags":["ssd","fast"]}]',
+                DEFAULT_NODE_TAG_ANNOTATION: '[",.*invalid-tag"]',
+            }
+        }
+    })
+    # Make sure Longhorn has tried to apply the annotation.
+    time.sleep(2)
+    node = client.by_id_node(node_name)
+    assert len(node.disks) == 0
+    assert not node.tags
+
+    disk = {"path": DEFAULT_DISK_PATH, "allowScheduling": True}
+    node.diskUpdate(disks=[disk])
+    node = wait_for_disk_update(client, node_name, 1)
+    assert len(node.disks) == 1
+    for _, disk in iter(node.disks.items()):
+        assert disk.path == DEFAULT_DISK_PATH
+        assert disk.allowScheduling is True
+        assert disk.storageReserved == 0
+        assert not disk.tags
+        break
+    client.update(node, tags=["tag1", "tag2"])
+    wait_for_node_tag_update(client, node_name, ["tag1", "tag2"])
+
+    # The existing node disks keep unchanged
+    # even if the annotation is corrected.
+    core_api.patch_node(node_name, {
+        "metadata": {
+            "annotations": {
+                DEFAULT_DISK_CONFIG_ANNOTATION:
+                    '[{"path":"' + DEFAULT_DISK_PATH +
+                    '","allowScheduling":false,' +
+                    '"storageReserved":2048,"tags":["hdd","slow"]}]',
+            }
+        }
+    })
+    time.sleep(2)
+    assert len(node.disks) == 1
+    for _, disk in iter(node.disks.items()):
+        assert disk.path == DEFAULT_DISK_PATH
+        assert disk.allowScheduling is True
+        assert disk.storageReserved == 0
+        assert not disk.tags
+        break
+
+    # Clean up all disks
+    # then the correct annotation should be applied.
+    node = client.by_id_node(node_name)
+    disks = node.disks
+    for _, disk in iter(disks.items()):
+        disk.allowScheduling = False
+    update_disks = get_update_disks(disks)
+    node = client.by_id_node(node_name)
+    node.diskUpdate(disks=update_disks)
+    node.diskUpdate(disks=[])
+    time.sleep(2)
```

Comment:
There is a wait below, why we need to use `time.sleep` here?

Reply:
Before the above disk cleanup, there is one existing disk on the node.
After the disk cleanup, node controller will re-create a new disk immediately according to the annotation.
Hence the below wait cannot guarantee that the cleanup and recreation are complete or not.

Similar below

Reply:
Then why sleeping 2 seconds can guarantee that?

Reply:
OK. It's better to use `wait_for_disk_status()` rather than sleep and `wait_for_disk_update`